src/Components/UI/Calculator.js

```diff
@@ -1,109 +1,108 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../Logic/calculate';
 import './Calculator.css';
 
-class Calculator extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: '',
-      next: null,
-      operation: null,
-    };
-  }
+const Calculator = () => {
+  const [total, setTotal] = useState('');
+  const [next, setNext] = useState('');
+  const [operation, setOperation] = useState('');
 
-  addInput = (event) => {
+  const addInput = (event) => {
     const clickedNumber = event.target.innerText;
-    const { total, next, operation } = this.state;
     if (operation === null && next === null) {
       if (total === 0 || total === '0') {
-        this.setState({ total: '' });
+        setTotal('');
       }
       const temp = total.toString() + clickedNumber.toString();
-      this.setState({ total: temp });
+      setTotal(temp);
     } else if (operation !== null && total !== null) {
       if (next === null) {
-        this.setState({ next: clickedNumber.toString() });
+        next(clickedNumber.toString());
       } else {
         const temp = next.toString() + clickedNumber.toString();
-        this.setState({ next: temp });
+        setNext(temp);
       }
     }
   };
 
-  addOperation = (event) => {
+  const addOperation = (event) => {
     const clickedOperation = event.target.innerText;
-    this.setState({ operation: clickedOperation.toString() });
+    setOperation(clickedOperation.toString());
   };
 
-  callEqual = (event) => {
+  const callEqual = (event) => {
     const clickedEqual = event.target.innerText;
-    const result = calculate(this.state, clickedEqual);
-    this.setState(result);
+    const result = calculate({ total, next, operation },
+      clickedEqual);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
   };
 
-  callAC = (event) => {
+  const callAC = (event) => {
     const clickedAC = event.target.innerText;
-    const result = calculate(this.state, clickedAC);
-    this.setState(result);
+    const result = calculate({ total, next, operation }, clickedAC);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
   };
 
-  callReverse = (event) => {
+  const callReverse = (event) => {
     const clickedReverse = event.target.innerText;
-    const result = calculate(this.state, clickedReverse);
-    this.setState(result);
+    const result = calculate({ total, next, operation }, clickedReverse);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
```

Comment:
- [ ] When trying to do any of the calculator operations nothing happened, So please fix that.

![chrome-capture-2022-9-28](https://user-images.githubusercontent.com/68796232/198581814-7fd3971c-2cc9-46ff-a559-1c761b48e0f1.gif)

Reply:
hi @Omar-Muhamad, can you try again please it is not like that on my computer.